rest-service/manager_rest/deployment_update/manager.py

```diff
@@ -251,6 +253,8 @@ def commit_deployment_update(self,
         self.sm.update(dep_update)
 
         # If this is a preview, no need to run workflow and update DB
+        deployment = self.sm.get(models.Deployment, dep_update.deployment_id)
+        labels_to_create = self._get_deployment_labels_to_create(dep_update)
```

Comment:
Any reason why you changed the location of these variables?

Reply:
yes, because I want to have the reference  for deployment so that I can use it when verify deployment label parent id but I think I can drop the verification because its already happened on the blueprint upload

Reply:
But then, if an invalid list of labels was provided, the line `self.sm.update(dep_update)` will still be called, and I don't think we want that.

Reply:
Aha yeah that is a valid point then I will keep it and make it before preview

Reply:
It is supposed to be before this line https://github.com/cloudify-cosmo/cloudify-manager/pull/2843/files#diff-e62ae4f438aaf3d7e37135a0065891a2814a47c4ffa74fa9077cfe049eaa3b08R253

Reply:
I think even before, there is another update before it

Reply:
I don't think so. This line defines that the current status is `STATES.UPDATING`, and it might be used afterwards if something failed.